fcos-policy-engine/src/utils.rs

```diff
@@ -0,0 +1,34 @@
+use actix::prelude::*;
+use commons::graph;
+use failure::{Error, Fallible};
+use reqwest::Method;
+use std::time::Duration;
+
+/// Default timeout for HTTP requests (30 minutes).
+const DEFAULT_HTTP_REQ_TIMEOUT: Duration = Duration::from_secs(30 * 60);
+/// Default address of fcos-graph-builder, which is the same as fcos-policy-builer
+const DEFAULT_GB_ADDR: &str = "http://127.0.0.1:5050/v1/graph";
+
+/// Return a request builder with base URL and parameters set.
+fn new_request(method: reqwest::Method, url: reqwest::Url) -> Fallible<reqwest::RequestBuilder> {
+    let client = reqwest::ClientBuilder::new()
+        .timeout(DEFAULT_HTTP_REQ_TIMEOUT)
+        .build()?;
+    let builder = client.request(method, url);
+    Ok(builder)
+}
+
+/// Fetch the graph from the fcos-graph-builder instance with the query specified.
+pub(crate) fn fetch_graph_from_gb(
+    query: String,
```

Comment:
It should be enough to pass the stream query parameter from the PE to the GB.
That is, passing a `stream` argument here and building the `stream=xyz` query internally in this function.
(I think there is also a missing check at the calling site to make sure the argument isn't an empty-string)

Reply:
And also the `basearch` value right? since it was also used in the graph builder.

Reply:
Absolutely yes, sorry I totally forgot about that one.